src/components/controls/language.js

```diff
@@ -16,8 +16,18 @@ import { CHANGE_LANGUAGE } from "../../actions/types";
   };
 })
 class Language extends React.Component {
+  async ensureLanguageResources(lang) {
+    for (const ns of ["language", "sidebar", "translation"]) {
+      if (!i18n.hasResourceBundle(lang, ns)) {
+        const res = await import(/* webpackMode: "lazy-once" */ `../../locales/${lang}/${ns}.json`); // eslint-disable-line
+        i18n.addResourceBundle(lang, ns, res.default);
+      }
+    }
+  }
 
-  componentWillMount() {
+  async componentWillMount() {
+    if (!this.props.language || this.props.language === "en") return;
```

Comment:
Not all translate keys return the same string value in english -- `t(x) === x` is mostly, but not always, true. I think the current implementation deliberately doesn't load the English local, instead utilizing the fact that i18n will fall back to the provided strings. This results in errors in only a few places, for instance clicking on the "Download data" modal produces 

![image](https://user-images.githubusercontent.com/8350992/83124727-ad838f00-a12a-11ea-97f1-f3145f216e12.png)

I think an acceptable solution would be to let the English locale be bundled wherever webpack wants to put it (it's <2kb gz) but keep the other languages in a separate bundle which is only fetched when necessary (as done now).

Reply:
Uh, I had completely missed this. Yes I think your solution is best, I modified a tiny bit index.js and webpack.config.js to accommodate for it.